ait/commons/util/command/upload.py

```diff
@@ -18,6 +21,54 @@ def __init__(self, aws, args):
         self.args = args
         self.files = []
 
+    def upload_file(self, f, key):
+        session = self.aws.new_session()
+        s3 = session.resource('s3')
+
+        ftype = filetype.guess(f)
+        # default contentType
+        content_type = 'application/octet-stream'
+        if ftype is not None:
+            content_type = ftype.mime
+        content_type += '; dcp-type=data'
+
+        s3.Bucket(self.aws.bucket_name).upload_file(Filename=f,
+                                                    Key=key,
+                                                    Callback=ProgressBar(target=f, total=os.path.getsize(f)),
+                                                    ExtraArgs={'ContentType': content_type}
+                                                    )
+        return True
+
+    def upload_files(self, files, prefix):
+
+        with ThreadPoolExecutor() as executor:
+            # submit each job and map future to file
+            futures = {}
+            for f in enumerate(files):
+                key = f"{prefix}{os.path.basename(f)}"
+                size = os.path.getsize(f)
+
+                if not self.args.o and self.aws.obj_exists(key):
+                    print(f"{f} already exists. Use -o to overwrite.")
+
+                elif size == 0:
+                    print(f"{f} is an empty file")
+
+                else:
+                    futures[executor.submit(self.upload_file, f, key)] = f
```

Comment:
For readability and terseness, I think you could use `concurrent.futures.Executor.map` here. You would have to move the file size and file existing checks into another function or into the `upload_file` function. E.g. `executor.map(self.upload_file, files)`

Reply:
That was more of a choice, I didn't want `upload_file` function to do the checks

Reply:
This would also help testability.

Reply:
You can move the checks into a validate function and then have upload_file call that, or use composition to call both from map

Reply:
Probably better if the validate function throws an error rather than just logging too